### PR TITLE
Dedicated packages for webhook handlers in GSAC

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area dev-productivity
/kind cleanup
/priority normal

**What this PR does / why we need it**:
Similar to #3644, for better isolation we should have dedicated package for the webhook handlers in the GSAC (gardener-seed-admission-controller).

**Which issue(s) this PR fixes**:
Follow-up after #3459

**Special notes for your reviewer**:
PR has several commits, so
/squash

**Release note**:

```noteworthy operator
NONE
```

### CODE REVIEW

File: pkg/seedadmission/webhooks/admission/extensioncrds/admission_test.go [hunk 1]
```diff
@@ -575,170 +576,21 @@ var _ = Describe("Extension Deletion Protection", func() {
 			})
 		})
 	})
-
-	Describe("Integration Test", func() {
```
Comment (timebertt) [inline pkg/seedadmission/webhooks/admission/extensioncrds/admission_test.go]: Why did you move this to the `suite_test.go` file?

**Review by timebertt — DISMISSED** [review]
/lgtm

File: pkg/seedadmission/webhooks/admission/extensioncrds/admission_test.go (hunk 1 above)
Comment (rfranzke) [reply pkg/seedadmission/webhooks/admission/extensioncrds/admission_test.go]: Hm, I would need the `restConfig` of the testEnv here in this files and I don't have access to it. Do you suggest to start dedicated test envs in the new admission packages instead of the "central" test env in `./pkg/seedadmissioncontroller`?

Comment (timebertt) [reply pkg/seedadmission/webhooks/admission/extensioncrds/admission_test.go]: Ah, of course. Makes perfect sense. I didn't notice the package move in the diff...
No, I would not start dedicated test envs for this.

What do you think about having separate files (like `extension_crds_test.go` and `podschedulername_test.go`) containing the different integration tests in the `seedadmissioncontroller` package then instead of having both tests in the `suite_test.go`?

Comment (rfranzke) [reply pkg/seedadmission/webhooks/admission/extensioncrds/admission_test.go]: Sounds like a good idea, will do!

Comment (rfranzke) [reply pkg/seedadmission/webhooks/admission/extensioncrds/admission_test.go]: Done in 295ba93 and df9953f.

I've also renamed `plugin` to `handler` in 11354c0.

**Review by timebertt — APPROVED** [review]
Nice, thank you!
/lgtm
